pandas/core/indexes/multi.py

```diff
@@ -1412,6 +1420,107 @@ def from_product(cls, iterables, sortorder=None, names=None):
         labels = cartesian_product(labels)
         return MultiIndex(levels, labels, sortorder=sortorder, names=names)
 
+    @classmethod
+    def from_frame(cls, df, sortorder=None, names=None, squeeze=True):
+        """
+        Make a MultiIndex from a DataFrame.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be converted to MultiIndex.
+        sortorder : int or None
+            Level of sortedness (must be lexicographically sorted by that
+            level).
+        names : list-like / callable, optonal
+            If no names provided, use column names, or tuple of column names if
```

Comment:
optional

If no names are provided, use the column names, or .....

If a squence, overwrite....

If a callable